### DISCUSSION TITLE
How to create a new session

### DISCUSSION THREAD

**Discussion by fritzrehde** [qutebrowser/qutebrowser#7699] [Q&A]
I'm not sure I understand the sessions feature in qutebrowser correctly. I currently use tmux, and like it a lot for managing terminal sessions, and would like to have something similar in my browser. Does qutebrowser offer similar functionality with its `session`s? I.e. I can create a new session, switch to/load a session, and quit a session? If yes, I can't find a "create session" or "new session" command, if I want to create a new empty session.

**Comment by The-Compiler** [comment]
While I use tmux (and have used its sessions feature before), I don't really understand what that would mean in the context of a browser. Can you elaborate on what you'd like to do exactly (i.e., what does creating/switching to/loading/quitting a session entail?)? Maybe you're looking for `--basedir`?

With qutebrowser sessions, a session is basically just a collection of open windows and their tabs (and scroll positions and such, but you get the idea).

**Reply by fritzrehde** [reply]
If I'm working on a specific (university/coding) project with 10+ tabs open, I might want to create a new session when I go into a lecture so I have a "fresh"/"clean" browser. Then, once I'm done with the lecture, I'd delete that session and switch back to the project session. That's the use-case I envision

**Reply by mschilli87** [reply]
@fritzrehde: I might just be missing something obvious, but what part of that use-case is not covered by `--temp-basedir` ([as previously suggested by @The-Compiler](https://github.com/qutebrowser/qutebrowser/discussions/7699#discussioncomment-5905231))?

**Reply by fritzrehde** [reply]
Oh, I never said it didn't cover my use-case, I just wanted to detail my use-case, I haven't tried `--temp-basedir` yet. Sorry for the confusion :)

**Reply by fritzrehde** [reply]
I'm still a little unsure how to use `--temp-basedir` and `--basedir`. Do you have an example how to use them in a config. Do I have to call `qutebrowser` with these commands?

**Reply by mschilli87** [reply]
@fritzrehde: Sorry for confusing you. I guess `--basedir` fits your use case much better:

1. Start qutebrowser with a project-specific basedir. (-> Create a project-specific session.)
2. Open your 10+ tabs.
3. Going into the lecture, close qutebrowser (saving the state to the project-specific basedir). (-> Detatch the project-specific session).
4. Open qutebrowser with a different basedir (for the lecture), `--tmp-basedir` (starting a throw-away session) or without `--[temp-]basedir` parameter (using the default session). (-> Create/Attach another (tempory/default) session.)
5. Do your work.
6. After the lecture, close qutebrowser. (-> Close (temporary) / Detach (named/default) session.)
7. Open qutebrowser with the project-specific basedir. (-> (Re-)attach the project-specific session.)

I suggested `--temp-basedir` as a shortcut if you don't need more than two sessions at once (*i.e.* there is only one project at a time):

1. Open qutebrowser (no `--[temp-]basedir` -> default session).
2. Do your (project related) work.
3. Going into the lecture, close qutebrowser (saving the state to the default basedir).
4. Open qutebrowser with `--temp-basedir`.
5. Do your work.
6. After the lecture, close qutebrowser.
7. Open qutebrowser (no `--[temp-]basedir`) to get back to the default session (that has your project-related tabs).

I hope this clarifies the situation a bit.

If you want to re-use (a lot of different) sessions, you could create aliases / key binding in your shell / WM for each of them to start qutebrowser with a specific `--basedir`. Not that nothing stops you from running several of them at once. So this is not a 1:1 with tmux.
